pyfstat/grid_based_searches.py

```diff
@@ -774,22 +778,32 @@ def run(self, return_data=False):
             )
         )
 
-        data = np.array(data, dtype=np.float)
         if return_data:
             return data
         else:
-            self.save_array_to_disk(data)
             self.data = data
+            self.save_array_to_disk()
+
+    def get_transient_fstat_map_filename(self, param_point):
```

Comment:
This was supposed to be its own commit, but probably too bothersome to extract now that I've somehow squashed it in... should amend commit message to acknowledge the addition, though.